sw/device/tests/hmac_enc_irq_test.c

```diff
@@ -0,0 +1,130 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/hmac_testutils.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+/**
+ * HMAC done 50uS timeout, which should be enough even for a 100kHz device.
+ *
+ * https://docs.opentitan.org/hw/ip/hmac/doc/
+ * Final hash calculation takes 360 cycles, which consists of one
+ * block compression and extra HMAC computation.
+ */
+const uint32_t kHmacEncFinishTimeoutUsec = 50;
+
+/**
+ * FIFO empty 10uS timeout, which should be enough even for a 100kHz device.
+ *
+ * https://docs.opentitan.org/hw/ip/hmac/doc/
+ * Single HMAC block compression takes 80 cycles.
+ */
+const uint32_t kHmacEncEmptyTimeoutUsec = 10;
+
+static plic_isr_ctx_t plic_ctx = {
+    .hart_id = kTopEarlgreyPlicTargetIbex0,
+};
+
+static dif_hmac_t hmac;
+static top_earlgrey_plic_peripheral_t peripheral_serviced;
+static dif_hmac_irq_t irq_serviced;
+static struct hmac_isr_ctx hmac_ctx = {
```

Comment:
```suggestion
static struct hmac_isr_ctx_t hmac_ctx = {
```
to match the fix you made to the autogen'd testutils?